test/integration/request-flow/stop-execution-from-sandbox.test.js

```diff
@@ -0,0 +1,291 @@
+                        }
+                    ],
+                    request: {
+                        url: 'https://postman-echo.com/get?param={{key-1}}:{{key-2}}:{{key-3}}:{{key-4}}',
+                        auth: {
+                            type: 'bearer',
+                            bearer: {
+                                token: '{{key-1}}:{{key-2}}:{{key-3}}:{{key-4}}'
+                            }
+                        }
+                    }
+                }]
+            }
+        });
+
+        this.run(clonedRunOptions, function (err, results) {
+            testRun = results;
+
+            done(err);
+        });
+    });
+
+    it('should not reflect any variable change line after pm.request.stopExecution in pre-request script', function () {
+        expect(testRun.console.callCount).to.equal(4);
+
+        var collPreConsole = testRun.console.getCall(0).args.slice(2),
+            collPreConsole2 = testRun.console.getCall(1).args.slice(2),
+            item2PreConsole = testRun.console.getCall(2).args.slice(2),
+            item2TestConsole = testRun.console.getCall(3).args.slice(2);
+
+        expect(testRun).to.be.ok;
+        expect(testRun).to.nested.include({
+            'done.calledOnce': true,
+            'start.calledOnce': true,
+            'io.calledOnce': true,
+            'request.calledOnce': true
+        });
+
+        expect(collPreConsole).to.deep.include.ordered.members([
+            'collection pre',
+            {
+                'key-1': 'global-value-1',
+                'key-2': 'coll-value-2',
+                'key-3': 'env-value-3',
+                'key-4': 'data-value-4'
+            }
+        ]);
+
+        expect(collPreConsole2).to.deep.include.ordered.members([
+            'collection pre',
+            {
+                'key-1': 'modified-1',
+                'key-2': 'coll-value-2',
+                'key-3': 'env-value-3',
+                'key-4': 'data-value-4'
+            }
+        ]);
+
+        expect(item2PreConsole).to.deep.include.ordered.members([
+            'item 2 pre',
+            {
+                'key-1': 'modified-1',
+                'key-2': 'coll-value-2',
+                'key-3': 'env-value-3',
+                'key-4': 'modified-4'
+            }
+        ]);
+
+        expect(item2TestConsole).to.deep.include.ordered.members([
+            'item 2 test',
+            {
+                'key-1': 'modified-1',
+                'key-2': 'coll-value-2',
+                'key-3': 'env-value-3',
+                'key-4': 'modified-4'
+            }
+        ]);
+    });
+
+    it('should send request with correct URL, variables after stopExecution should not be reflected', function () {
+        var url1 = testRun.request.getCall(0).args[3].url.toString(),
+            expectedToken = 'modified-1:coll-value-2:env-value-3:modified-4';
+
+        expect(url1).to.equal('https://postman-echo.com/get?param=' + expectedToken);
+    });
```

Comment:
Let's test other apis that can cause side effects:
- `.setNextRequest`
- `.sendRequest`
- request auths which log to console.

Reply:
Added for setNextRequest and sendRequest